tls/s2n_handshake_io.c

```diff
@@ -850,12 +850,36 @@ static int s2n_handshake_handle_sslv2(struct s2n_connection *conn)
 
     /* Handle an SSLv2 client hello */
     GUARD(s2n_stuffer_copy(&conn->in, &conn->handshake.io, s2n_stuffer_data_available(&conn->in)));
-    GUARD(s2n_sslv2_client_hello_recv(conn));
+    /* Set the client hello version */
+    conn->client_hello_version = S2N_SSLv2;
+    /* Execute the state machine handler */
+    int r = ACTIVE_STATE(conn).handler[conn->mode](conn);
```

Comment:
We talked about this offline, and I'm not sure we have a good answer. This function is copying a lot of logic from the main handshake functions, but it's unclear how to merge it back in. :\

Reply:
will create an issue

Reply:
https://github.com/awslabs/s2n/issues/1601